src/main/java/com/example/demo/PatientController.java

```diff
@@ -0,0 +1,53 @@
+package com.example.demo;
+
+import org.springframework.web.bind.annotation.*;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/patients")
+public class PatientController {
+    private List<Patient> patients = new ArrayList<>();
+
+    @GetMapping
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    @GetMapping("/{email}")
+    public Patient getPatientByEmail(@PathVariable String email) {
+        Optional<Patient> findPatient = patients.stream()
+                .filter(patient -> patient.getEmail().equals(email))
```

Comment:
.filter(patient -> patient.getEmail().equals(email)) jesli nie mamy pewnosci ze patient posiada email to powinnismy:
.filter(patient -> patient.getEmail() != null && patient.getEmail().equals(email)) lub
.filter(patient -> Objects.equals(patient.getEmail(), email));****